test/cluster.go

```diff
@@ -80,26 +90,29 @@ func AssertEqualClusterDetails(t *testing.T, expected, actual *repository.Cluste
 	assert.Equal(t, expected.CapacityExhausted, actual.CapacityExhausted)
 }
 
-func AssertEqualClusterData(t *testing.T, clusters []repository.Cluster, clusterList []*app.ClusterData) {
-	require.Len(t, clusterList, len(clusters))
-
-	for _, c := range clusterList {
+// AssertEqualClustersData verifies that data for all actual clusters match the expected ones
+func AssertEqualClustersData(t *testing.T, expected []repository.Cluster, actual []*app.ClusterData) {
+	require.Len(t, actual, len(expected))
+	for _, c := range actual {
 		require.NotNil(t, c)
-		apiURL := c.APIURL
-		require.NotNil(t, apiURL)
-		cluster := FilterClusterByURL(apiURL, clusters)
-		require.NotNil(t, cluster, "cluster with url %s could not found", apiURL)
-		assert.Equal(t, cluster.Name, c.Name)
-		assert.Equal(t, cluster.URL, apiURL)
-		assert.Equal(t, cluster.ConsoleURL, c.ConsoleURL)
-		assert.Equal(t, cluster.MetricsURL, c.MetricsURL)
-		assert.Equal(t, cluster.LoggingURL, c.LoggingURL)
-		assert.Equal(t, cluster.AppDNS, c.AppDNS)
-		assert.Equal(t, cluster.Type, c.Type)
-		assert.Equal(t, cluster.CapacityExhausted, c.CapacityExhausted)
+		e := FilterClusterByURL(c.APIURL, expected)
+		require.NotNil(t, e, "cluster with url %s could not found", c.APIURL)
+		AssertEqualClusterData(t, e, c)
 	}
 }
 
+// AssertEqualClusterData verifies that data for actual cluster match the expected one
+func AssertEqualClusterData(t *testing.T, expected *repository.Cluster, actual *app.ClusterData) {
+	assert.Equal(t, expected.Name, actual.Name)
+	assert.Equal(t, httpsupport.AddTrailingSlashToURL(expected.URL), actual.APIURL)
```

Comment:
Wouldn't it be safer to check URLs as is instead of adding trailing slashes to expected URLs?
We should be really careful with these trailing slashes. It's easy to miss it.

Reply:
done in 238c902 with a `Normalize` function outside of the `Assert..` funcs